### PR TITLE
feat: add http and grpc API call logs

### PR DESCRIPTION
**Pull request by bap2pecs**
## Summary

while debugging the issue where the L2 finalized head stuck due to a potential reorg, I found that we don't have server logs in the FG grpc server. So I don't know when op-node calls it with what data

so let's add logs for all API calls, for both http and grpc servers

this is built on top of https://github.com/babylonlabs-io/finality-gadget/pull/42

## Test Plan

```
make lint
make test
```

both passed

### CODE REVIEW

File: server/grpcserver.go [hunk 1]
```diff
@@ -2,13 +2,21 @@ package server
...
+	"fmt"
 
 	"github.com/babylonlabs-io/finality-gadget/proto"
 	"github.com/babylonlabs-io/finality-gadget/types"
+	"go.uber.org/zap"
 )
 
 // QueryIsBlockBabylonFinalized is an RPC method that returns the finality status of a block by querying the internal db.
 func (s *Server) QueryIsBlockBabylonFinalized(ctx context.Context, req *proto.QueryIsBlockBabylonFinalizedRequest) (*proto.QueryIsBlockFinalizedResponse, error) {
+	s.logger.Info(
```
Comment (parketh) [inline server/grpcserver.go:14]: imo we should change these to debug logs, these grpc calls happen quite often so can crowd out other logs if always enabled

Comment (SebastianElvis) [reply server/grpcserver.go:14]: Agreed, and the message can be a bit more informative, e.g., `QueryIsBlockBabylonFinalized` -> `QueryIsBlockBabylonFinalized is called`

**Review by SebastianElvis — APPROVED** [review]

**Comment by bap2pecs** [comment]
seems CI was failing before this PR. need to investigate

image

**Comment by bap2pecs** [comment]
will fix test together after the FP e2e test is fixed

**Pull request merged**